app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java

```diff
@@ -336,23 +353,145 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent)
+
+                                @Override
+                                public void onResponse(JSONObject response)
+                                {
+                                    parseBookResult(response);
+                                }
+                            }, new Response.ErrorListener()
+                            {
+                                @Override
+                                public void onErrorResponse(VolleyError error)
+                                {
+                                /*
+                                   In this case, we simply won't add the additional book details
+                                */
+                                    Log.d("Volley Error: ", error.getMessage());
+                                }
+                            });
+                    requestQueue.add(jsonObjectRequest);
+                }
+                else
+                {
+                    super.onActivityResult(requestCode, resultCode, intent);
                 }
-                isbnView.setText(isbn_scan_result);
             }
-            else
+        }
+
+    }
+
+    /**
+     * Parse the result received from Google's book API after scanning an ISBN
+     */
+    private void parseBookResult(JSONObject result)
+    {
+        String title = "";
+        String subtitle = "";
+        String author = "";
+        String description = "";
+        String imageUrlString = "";
+        URL imageUrl = null;
+        try
+        {
+            /* If there are no matches, return */
+            if (result.getInt("totalItems") < 1)
             {
-                super.onActivityResult(requestCode, resultCode, intent);
+                return;
             }
+
+            /* Multiple books may be returned by the books API, select the first one */
+            JSONObject volumeInfo = result.getJSONArray("items").getJSONObject(0).getJSONObject("volumeInfo");
+            JSONArray authors = volumeInfo.getJSONArray("authors");
+            /* Since there may be multiple authors, append them together separated by a comma */
+            for (int i = 0; i < authors.length(); i++)
+            {
+                if (i == 0)
+                {
+                    author += authors.getString(i);
+                }
+                else
+                {
+                    author += ", " + authors.getString(i);
+                }
+            }
+
+            /*
+               The subtitle may not exist. Catch it in here so that the exception doesn't bubble up
+               and cause no data to be set for the book.
+             */
+            try
+            {
+                subtitle = volumeInfo.getString("subtitle");
+            } catch (JSONException e)
+            {
+                /* Do nothing, this is expected to happen sometimes */
+            }
+            title = (subtitle == "") ? (volumeInfo.getString("title")) : (volumeInfo.getString("title") + ": " + subtitle);
```

Comment:
Maybe check for each individual field before assignment? It'd be nice to have partial information rather than fail everything when one field isn't valid.